docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,421 @@
+
+When the sharded state dict is provided as input, it is processed in the following way (see also ``load`` function `documentation <https://docs.nvidia.com/megatron-core/developer-guide/latest/api-guide/dist_checkpointing.html#module-core.dist_checkpointing.serialization>`_):
+
+1. The "common" state dict is loaded from the checkpoint. This forms the base of the resulting state dict.
+2. The ShardedTensorFactories from the input sharded state dict are applied.
+3. The LocalNonPersistentObjects are extracted from the input sharded state dict, unwrapped and added to the resulting state dict.
+4. The ShardedObjects are extracted and loaded from the checkpoint into the resulting state dict.
+5. The ShardedTensors are extracted and loaded from the checkpoint into the resulting state dict.
+6. Factory merges are applied (see `Optimizers`_ for explanation).
+
+This results in a *regular* state dict with plain tensors that can be further processed by the application (which usually means running ``model.load_state_dict(state_dict)``.
+
+
+dist_checkpointing.load_common_state_dict
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_common_state_dict`` function is an entry point that allows loading only the “common” part of the checkpoints.
+Most of the checkpoint config and metadata can be loaded with this method, which allows skipping data loading in order to take decisions regarding checkpoint config, version, etc.
+
+dist_checkpointing.load_tensors_metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_tensors_metadata`` function is an entry point that allows reading all ShardedTensors metadata from the checkpoint without loading any data.
+The result is a sharded state dict with trivial sharding (every tensor is sharded into one big shard).
+
+dist_checkpointing.load_plain_tensors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_plain_tensors`` function is an entry point that allows reading sharded tensors stored in the checkpoint without any sharding (as plain tensors).
+This function is simply a composition of ``load_tensors_metadata`` and ``save``.
+
+Save and Load Strategies
+------------------------
+There are multiple ways to save a sharded state dict into a serialized checkpoint. They can be provided by the user as saving and loading strategies.
+
+There are four types of strategies:
+
+1. Saving strategy for ShardedTensors
+2. Saving strategy for "common" data
+3. Loading strategy for ShardedTensors
+4. Loading strategy for "common" data
+
+Additionally, ShardedObjects are handled with either "sharded" or "common" strategy depending on its capabilities (``can_handle_sharded_objects`` property).
+
+Each saving strategy is associated with a ``backend`` and a ``version``.
+Each loading strategy can be associated with multiple ``backend``s and ``version``s it can load.
```

Comment:
text inside backticks doesn't render correctly due to "s" after it.

Maybe:
"Each loading strategy can be associated with multiple values of ``backend`` and ``version``." ?

Reply:
or just don't use backticks in this sentence (since you don't use them in the next sentence anyway. or instead, maybe should use backticks in all sentences)